backend/api_calls.py

```diff
@@ -877,6 +877,168 @@ def RandomWords(n):
   return '-'.join(words)
 
 
+def SendNotification(request, firebase):
+  """Queue a notification to be sent.
+
+  Validation:
+    notificationId does not exist.
+    groupId or playerId exists.
+
+  Args:
+    notificationId: Queued Notification id to create.
+    message: Message to send to client.
+    previewMessage: Short preview of the message conents. Optional.
+    app: Whether to send as a push notification to the iOS/Android/Web clients.
+    vibrate: Whether the notification should vibrate on iOS/Android.
+    sound: Whether the notification should play a sound on iOS/Android.
+    destination: URL that notification should open when clicked.
+    sendTime: When to spawn notifications from this template. UNIX timestamp.
+    playerId: Player id to send message to.
+    groupId: Who to send the notifications to.
+    icon: An icon code to show.
+
+  Firebase entries:
+    /notifications/%(notificationId)
+"""
+  valid_args = ['!notificationId']
+  if 'groupId' in request:
+    valid_args.append('groupId')
+  elif 'playerId' in request:
+    valid_args.append('playerId')
+  else:
+    raise InvalidInputError('Must include either a playerId or a groupId')
```

Comment:
I would put both into the valid_args to keep that simple then be explicit about that test.
if 'groupId' in request and 'playerId' in request:
  raise InvalidInput('You cannot do both')